libs/src/api/user.js

```diff
@@ -175,9 +190,99 @@ class Users extends Base {
+  }) {
+    this.REQUIRES(Services.CREATOR_NODE)
+    const phases = {
+      CLEAN_AND_VALIDATE_METADATA: 'CLEAN_AND_VALIDATE_METADATA',
+      AUTOSELECT_CONTENT_NODES: 'AUTOSELECT_CONTENT_NODES',
+      SYNC_ACROSS_CONTENT_NODES: 'SYNC_ACROSS_CONTENT_NODES',
+      SET_PRIMARY: 'SET_PRIMARY',
+      UPLOAD_METADATA_AND_UPDATE_ON_CHAIN: 'UPLOAD_METADATA_AND_UPDATE_ON_CHAIN'
+    }
+    let phase = ''
+    const numNodes = 3
+
+    const user = this.userStateManager.getCurrentUser()
+    const newMetadata = { ...user }
+    // Failed the addUser() step
+    if (!user) { throw new Error('No current user') }
+    // No-op if the user already has a replica set assigned under creator_node_endpoint
+    if (user.creator_node_endpoint && user.creator_node_endpoint.length > 0) return
+
+    // The new metadata object that will contain the replica set
+    try {
+      // Create starter metadata and validate
+      phase = phases.CLEAN_AND_VALIDATE_METADATA
+
+      // Autoselect a new replica set and update the metadata object with new content node endpoints
+      let primary, secondaries
+      if (!newContentNodeEndpoints || newContentNodeEndpoints.length === 0) {
+        phase = phases.AUTOSELECT_CONTENT_NODES
+        const response = await serviceProvider.autoSelectCreatorNodes({
+          performSyncCheck: false,
+          whitelist: passList,
+          blacklist: blockList
+        })
+        primary = response.primary
+        secondaries = response.secondaries
+        if (!primary || !secondaries || secondaries.length < numNodes - 1) {
+          throw new Error(`Could not select a primary=${primary} and/or ${numNodes - 1} secondaries=${secondaries}`)
+        }
+        newContentNodeEndpoints = CreatorNode.buildEndpoint(primary, secondaries)
+      }
+      newMetadata.creator_node_endpoint = newContentNodeEndpoints
+
+      // Update the new primary to the auto-selected primary
+      phase = phases.SET_PRIMARY
+      const newPrimary = CreatorNode.getPrimary(newContentNodeEndpoints)
+      await this.creatorNode.setEndpoint(newPrimary)
+
+      // In signUp(), a replica set is assigned before uploading profile images.
+      // Should associate new metadata after profile photo upload.
+      phase = phases.UPLOAD_METADATA_AND_UPDATE_ON_CHAIN
+      await this._handleMetadata({
+        newMetadata,
+        userId
+      })
+    } catch (e) {
+      throw new Error(`assignReplicaSet() Error -- Phase ${phase}: ${e}`)
+    }
+
+    return newMetadata
+  }
+
+  /**
+   * Wrapper around private method _handleMetadata()
+   * @param {number} userId
+   * @param {Object} metadata
+   */
+  async updateUserMetadata (userId, metadata) {
+    await this._handleMetadata({
+      newMetadata: metadata,
+      userId
+    })
+    return userId
+  }
+
   /**
    * Util to upload profile picture and cover photo images and update
-   * a metadata object
+   * a metadata object. This method inherently calls triggerSecondarySyncs().
```

Comment:
signUp actually always relied on `uploadProfileImages` [code](https://github.com/AudiusProject/audius-protocol/blob/aef93ee8e79295655b217349a854123e59ac8943/libs/src/api/account.js#L150-L178), but I do agree that we could have some internal checks to make sure the writes actually successfully completed. 

a method name change for `_handleMetadata()` is a good idea. how about `_uploadAndUpdateMetadata()`?

Reply:
I don't know if we should add internal checks for photo upload in this PR as this feels a bit scope creep-y. we haven't done this for the prior upload flow and things have been ok since then

Reply:
agreed on last part - lets not add those safeguards right now given that this is not new